tests/controllers/feature/feature_test.go

```diff
@@ -0,0 +1,204 @@
+package feature_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/feature"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/wrangler/v2/pkg/crd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+type FeatureTestSuite struct {
+	suite.Suite
+	ctx             context.Context
+	cancel          context.CancelFunc
+	testEnv         *envtest.Environment
+	wranglerContext *wrangler.Context
+}
+
+var (
+	harvesterFeatureNoAnnotation = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester-baremetal-container-workload",
+		},
+	}
+	harvesterFeature = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester-baremetal-container-workload",
+			Annotations: map[string]string{
+				v3.ExperimentalFeatureValue: v3.ExperimentalFeatureKey,
+			},
+		},
+	}
+	//t = true
+	//f = false
+	h = v3.Feature{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester",
+		},
+	}
+)
+
+func (s *FeatureTestSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.TODO())
+	// Load CRD from YAML for REST Client
+	s.testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "..", "..", "pkg", "crds", "yaml", "generated", "management.cattle.io_features.yaml"),
+			filepath.Join("..", "..", "..", "pkg", "crds", "yaml", "generated", "management.cattle.io_nodedrivers.yaml"),
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+	restCfg, err := s.testEnv.Start()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), restCfg)
+
+	// Create CRDs
+	factory, err := crd.NewFactoryFromClient(restCfg)
+	assert.NoError(s.T(), err)
+	err = factory.BatchCreateCRDs(s.ctx,
+		crd.CRD{
+			SchemaObject: v3.Feature{},
+			NonNamespace: true,
+		},
+		crd.CRD{
+			SchemaObject: v3.NodeDriver{},
+			NonNamespace: true,
+		},
+	).BatchWait()
+	assert.NoError(s.T(), err)
+
+	// Create the wrangler context
+	s.wranglerContext, err = wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	// Register the feature controller
+	feature.Register(s.ctx, s.wranglerContext)
+
+	// Create and start the feature controller factory
+	sc := s.wranglerContext.ControllerFactory.ForResourceKind(schema.GroupVersionResource{
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "features",
+	}, "Feature", false)
+	assert.NoError(s.T(), err)
+	err = sc.Start(s.ctx, 1)
+	assert.NoError(s.T(), err)
+
+	s.wranglerContext.ControllerFactory.SharedCacheFactory().StartGVK(s.ctx, schema.GroupVersionKind{
+		Group:   "management.cattle.io",
+		Version: "v3",
+		Kind:    "nodedrivers",
+	})
```

Comment:
The problem is that the cache for `NodeDriver` is not initialized, and lasso is ignoring the `StartGVK` call. It retrurns [here](https://github.com/rancher/lasso/blob/master/pkg/cache/sharedinformerfactory.go#L83). We should call `ForKind` first in order to initialize the `NodeDriver` cache.

```suggestion
	_, err = s.wranglerContext.ControllerFactory.SharedCacheFactory().ForKind(schema.GroupVersionKind{
		Group:   "management.cattle.io",
		Version: "v3",
		Kind:    "NodeDriver",
	})
	assert.NoError(s.T(), err)

	err = s.wranglerContext.ControllerFactory.SharedCacheFactory().StartGVK(s.ctx, schema.GroupVersionKind{
		Group:   "management.cattle.io",
		Version: "v3",
		Kind:    "NodeDriver",
	})
```